code/modules/lighting/lighting_object.dm

```diff
@@ -103,46 +86,27 @@
 	#endif
 
 	if((rr & gr & br & ar) && (rg + gg + bg + ag + rb + gb + bb + ab == 8))
-	//anything that passes the first case is very likely to pass the second, and addition is a little faster in this case
-		icon_state = "transparent"
-		color = null
+		//anything that passes the first case is very likely to pass the second, and addition is a little faster in this case
+		affected_turf.underlays -= current_underlay
+		current_underlay.icon_state = "transparent"
```

Comment:
....... "transparent"?

Reply:
yes, lighting objects have one icon_state for when its completely bright (which is "transparent") and one when its completely dark ("dark"). if it's in-between it uses the null icon_state that uses a color matrix

Reply:
ok thank god